Failed: MapWithResourceSpec.MapWithResource must fail when close throws exception

```
sbt.ForkMain$ForkError: java.lang.AssertionError: assertion failed: expected OnNext(a), found OnError(akka.stream.testkit.Utils$TE: 
)
	at scala.Predef$.assert(Predef.scala:279)
	at akka.testkit.TestKitBase.expectMsg_internal(TestKit.scala:462)
	at akka.testkit.TestKitBase.expectMsg(TestKit.scala:438)
	at akka.testkit.TestKitBase.expectMsg$(TestKit.scala:438)
	at akka.testkit.TestKit.expectMsg(TestKit.scala:973)
	at akka.stream.testkit.TestSubscriber$ManualProbe.expectNext(StreamTestKit.scala:399)
	at akka.stream.scaladsl.MapWithResourceSpec.$anonfun$new$33(MapWithResourceSpec.scala:247)
	at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
	at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
	at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
	at org.scalatest.Transformer.apply(Transformer.scala:22)
	at org.scalatest.Transformer.apply(Transformer.scala:20)
	at org.scalatest.wordspec.AnyWordSpecLike$$anon$3.apply(AnyWordSpecLike.scala:1076)
	at org.scalatest.TestSuite.withFixture(TestSuite.scala:196)
	at org.scalatest.TestSuite.withFixture$(TestSuite.scala:195)
	at akka.stream.testkit.StreamSpec.withFixture(StreamSpec.scala:36)
	at org.scalatest.wordspec.AnyWordSpecLike.invokeWithFixture$1(AnyWordSpecLike.scala:1074)
	at org.scalatest.wordspec.AnyWordSpecLike.$anonfun$runTest$1(AnyWordSpecLike.scala:1086)
	at org.scalatest.SuperEngine.runTestImpl(Engine.scala:306)
	at org.scalatest.wordspec.AnyWordSpecLike.runTest(AnyWordSpecLike.scala:1086)
	at org.scalatest.wordspec.AnyWordSpecLike.runTest$(AnyWordSpecLike.scala:1068)
	at akka.testkit.AkkaSpec.runTest(AkkaSpec.scala:54)
	at org.scalatest.wordspec.AnyWordSpecLike.$anonfun$runTests$1(AnyWordSpecLike.scala:1145)
	at org.scalatest.SuperEngine.$anonfun$runTestsInBranch$1(Engine.scala:413)
	at scala.collection.immutable.List.foreach(List.scala:333)
	at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:401)
	at org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:390)
	at org.scalatest.SuperEngine.$anonfun$runTestsInBranch$1(Engine.scala:427)
	at scala.collection.immutable.List.foreach(List.scala:333)
	at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:401)
	at org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:396)
	at org.scalatest.SuperEngine.runTestsImpl(Engine.scala:475)
	at org.scalatest.wordspec.AnyWordSpecLike.runTests(AnyWordSpecLike.scala:1145)
	at org.scalatest.wordspec.AnyWordSpecLike.runTests$(AnyWordSpecLike.scala:1144)
	at akka.testkit.AkkaSpec.runTests(AkkaSpec.scala:54)
	at org.scalatest.Suite.run(Suite.scala:1112)
	at org.scalatest.Suite.run$(Suite.scala:1094)
	at akka.testkit.AkkaSpec.org$scalatest$wordspec$AnyWordSpecLike$$super$run(AkkaSpec.scala:54)
	at org.scalatest.wordspec.AnyWordSpecLike.$anonfun$run$1(AnyWordSpecLike.scala:1190)
	at org.scalatest.SuperEngine.runImpl(Engine.scala:535)
	at org.scalatest.wordspec.AnyWordSpecLike.run(AnyWordSpecLike.scala:1190)
	at org.scalatest.wordspec.AnyWordSpecLike.run$(AnyWordSpecLike.scala:1188)
	at akka.testkit.AkkaSpec.org$scalatest$BeforeAndAfterAll$$super$run(AkkaSpec.scala:54)
	at org.scalatest.BeforeAndAfterAll.liftedTree1$1(BeforeAndAfterAll.scala:213)
	at org.scalatest.BeforeAndAfterAll.run(BeforeAndAfterAll.scala:210)
	at org.scalatest.BeforeAndAfterAll.run$(BeforeAndAfterAll.scala:208)
	at akka.testkit.AkkaSpec.run(AkkaSpec.scala:54)
	at org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:318)
	at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:513)
	at sbt.ForkMain$Run.lambda$runTest$1(ForkMain.java:413)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

https://github.com/akka/akka/runs/8341017910

--- a/akka-stream-tests/src/test/scala/akka/stream/scaladsl/MapWithResourceSpec.scala
+++ b/akka-stream-tests/src/test/scala/akka/stream/scaladsl/MapWithResourceSpec.scala
@@ -236,16 +236,20 @@ class MapWithResourceSpec extends StreamSpec(UnboundedMailboxConfig) {
     }
 
     "fail when close throws exception" in {
-      val out = TestSubscriber.probe[String]()
-      Source
-        .single(1)
+      val (pub, sub) = TestSource
+        .probe[Int]
         .mapWithResource(() => Iterator("a"))((it, _) => if (it.hasNext) Some(it.next()) else None, _ => throw TE(""))
         .collect { case Some(line) => line }
-        .runWith(Sink.fromSubscriber(out))
+        .toMat(TestSink.probe)(Keep.both)
+        .run()
 
-      out.request(61)
-      out.expectNext("a")
-      out.expectError(TE(""))
+      pub.ensureSubscription()
+      sub.ensureSubscription()
+      sub.request(1)
+      pub.sendNext(1)
+      sub.expectNext("a")
+      pub.sendComplete()
+      sub.expectError(TE(""))
     }
 
     "not close the resource twice when read fails" in {

